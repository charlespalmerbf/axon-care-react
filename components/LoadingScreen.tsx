export default function LoadingScreen() {
  return (
    <div className="flex justify-center items-center w-full h-screen">
      <div className="flex items-center space-x-2">
        <div className="w-4 h-4 bg-green-500 rounded-full animate-bounce"></div>
        <div className="w-4 h-4 bg-green-500 rounded-full animate-bounce delay-150"></div>
        <div className="w-4 h-4 bg-green-500 rounded-full animate-bounce delay-300"></div>
      </div>
    </div>
  );
}
