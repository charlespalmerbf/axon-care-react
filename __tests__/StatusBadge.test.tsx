import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import { StatusBadge } from "@/components/StatusBadge";

describe("StatusBadge Component", () => {
  it("matches snapshot", () => {
    const screen = render(<StatusBadge status="scheduled" />);
    expect(screen).toMatchSnapshot();
  });

  it("renders with 'scheduled' status", () => {
    render(<StatusBadge status="scheduled" />);
    
    const badge = screen.getByText("scheduled");
    expect(badge).toBeInTheDocument();
    expect(badge).toHaveClass("text-green-500");

    const badgeContainer = badge.closest("div");
    expect(badgeContainer).toHaveClass("bg-green-600");

    const icon = screen.getByAltText("doctor");
    expect(icon).toHaveAttribute("src", expect.stringContaining("/assets/icons/check.svg"));
  });

  it("renders with 'pending' status", () => {
    render(<StatusBadge status="pending" />);
    
    const badge = screen.getByText("pending");
    expect(badge).toBeInTheDocument();
    expect(badge).toHaveClass("text-blue-500");

    const badgeContainer = badge.closest("div");
    expect(badgeContainer).toHaveClass("bg-blue-600");

    const icon = screen.getByAltText("doctor");
    expect(icon).toHaveAttribute("src", expect.stringContaining("/assets/icons/pending.svg"));
  });

  it("renders with 'cancelled' status", () => {
    render(<StatusBadge status="cancelled" />);
    
    const badge = screen.getByText("cancelled");
    expect(badge).toBeInTheDocument();
    expect(badge).toHaveClass("text-red-500");

    const badgeContainer = badge.closest("div");
    expect(badgeContainer).toHaveClass("bg-red-600");

    const icon = screen.getByAltText("doctor");
    expect(icon).toHaveAttribute("src", expect.stringContaining("/assets/icons/cancelled.svg"));
  });
});
