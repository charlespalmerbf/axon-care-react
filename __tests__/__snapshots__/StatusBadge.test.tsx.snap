// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBadge Component matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="status-badge bg-green-600"
      >
        <img
          alt="doctor"
          class="h-fit w-3"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/assets/icons/check.svg"
          style="color: transparent;"
          width="24"
        />
        <p
          class="text-12-semibold capitalize text-green-500"
        >
          scheduled
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="status-badge bg-green-600"
    >
      <img
        alt="doctor"
        class="h-fit w-3"
        data-nimg="1"
        decoding="async"
        height="24"
        loading="lazy"
        src="/assets/icons/check.svg"
        style="color: transparent;"
        width="24"
      />
      <p
        class="text-12-semibold capitalize text-green-500"
      >
        scheduled
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
